src/site/topic16.rst

```diff
@@ -1,8 +1,57 @@
-
 ***********************
 Topic #16 --- Iterators
 ***********************
 
+* Often we need to access every element in some *thing*
+    * like an array, ``Stack``, ``Queue``, or ``Bag``
+* We call this *iterating* over the things
+
+* We've done this countless times already with something like an array
+
+.. code-block:: Java
+    :linenos:
+
+    for(int i = 0; i < someArray.length; ++i) {
+        someMethod(someArray[i]);
+    }
+
+* However, not everything we may want to iterate over is an array, and trying to use the for loop with a linked structure won't work
+
+* The good news is, Java provides us with a common, uniform way to iterate over something
+* It is also independent of the implementation of the thing we want to iterate over
+
+
+Iterators
+=========
+
+
+Iterator Interface
+------------------
+
+
+Array Iterator
+--------------
+
+
+Linked Iterator
+---------------
+
+
+Collection Iterators
+====================
+
+
+toString
+--------
+
+
+Iterable
+========
```

Comment:
Word of warning once you explain this, `Iteratable` added `spliterator()` which is for parallel stream processing, they may be out of scope. That said, stream processing is super powerful, and leads people toward functional programming in general, which is all the good that's left in the world. Maybe worth an aside?

Reply:
Despite being out of scope and only seeing stream processing once in undergrad, 100% of the coding interviews I had dealt with streams. Also, functional is fun AF. 

In other words, amazing idea for an aside.